File: src/main/java/com/github/nsc/de/shake/parser/node/variables/VariableDivAssignmentNode.java
package com.github.nsc.de.shake.parser.node.variables;

import com.github.nsc.de.shake.parser.node.Node;
import com.github.nsc.de.shake.parser.node.ValuedNode;

public class VariableDivAssignmentNode implements ValuedNode {

    private final ValuedNode variable;
    private final Node value;
    private final int operatorPosition;

    public VariableDivAssignmentNode(ValuedNode variable, Node value, int operatorPosition) {
        this.variable = variable;
        this.value = value;
        this.operatorPosition = operatorPosition;
    }

    public ValuedNode getVariable() {
        return variable;
    }
    public Node getValue() {
        return value;
    }
    public int getOperatorPosition() {
        return operatorPosition;
    }

    @Override
    public String toString() {
        return '{' + getValue().toString() + "/=" + this.getValue().toString() + '}';
    }
}

File: parser/src/commonMain/kotlin/com/github/shakelang/shake/parser/node/variables/VariableDivAssignmentNode.kt
package com.github.shakelang.shake.parser.node.variables

import com.github.shakelang.shake.parser.node.*
import com.github.shakelang.shake.util.characterinput.position.PositionMap

class VariableDivAssignmentNode(
    map: PositionMap,
    val variable: ValuedNode,
    val value: Node,
    val operatorPosition: Int
) : ValuedNode(map) {

    override fun toJson(): Map<String, *> = mapOf(
        "name" to "VariableDivAssignmentNode",
        "variable" to variable.json,
        "value" to value.json
    )

}